### PR TITLE
feature: Add "last seen" info for lines with no trains

### PR DESCRIPTION
**Pull request by idreyn**
So thanks to @mathcolo and his work in #34, I had a good base to work from in implementing "last seen" information for lines that currently have no new trains running. It's not fancy, but it looks like this:

To fully configure this in production, we'll need to add

```
npm run create-history-db
```
to our deploy process, and then
```
npm run update-history-db
```
to our crontab or something to run periodically.

 _Test plan:_
- Run `npm run create-history-db`
- Update `npm run update-history-db` so that the update function is called with `test_mode=True` and then run `npm run update-history-db` to populate the database with some info, even if there are currently no new trains
- Tweak `initial_request_data()` to pass `test_mode=True` to `get_recent_sightings_for_lines`
- Open the train tracker and verify that for lines without new trains, you see a message about when a train (not actually new in this case) was actually seen on this line.

### CODE REVIEW

File: history/util.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import psycopg2
+
+import server.secrets as secrets
+
+HISTORY_TABLE_NAME = "newtrains_history"
+
+HISTORY_DB_SCHEMA = """
+CREATE TABLE IF NOT EXISTS newtrains_history (
+    id UUID DEFAULT uuid_generate_v4(),
```
Comment (mathcolo) [inline history/util.py:9]: This better than the nextval that's currently there?

File: history/create_history_db.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+def create_history_db():
+    cxn = get_history_db_connection(False)
+    cxn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+    cur = cxn.cursor()
+    cur.execute(f"CREATE DATABASE {POSTGRES_DB}")
+    cxn.close()
+    cxn = get_history_db_connection()
+    cxn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+    cur = cxn.cursor()
+    cur.execute('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";')
```
Comment (mathcolo) [inline history/create_history_db.py:16]: I assume this is for the id column? See comment below

File: server/secrets.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
-MBTA_V3_API_KEY = ''
-POSTGRES_DB = 'newtrains'
-POSTGRES_USER = 'ubuntu'  # No password; unix account auth
-'''
+MBTA_V3_API_KEY = ""
+POSTGRES_DB = "newtrains"
+POSTGRES_USER = "ian"  # No password; unix account auth
```
Comment (mathcolo) [inline server/secrets.py:3]: Should this not be ubuntu?

**Review by mathcolo — APPROVED** [review]
Awesome! few comments, but otherwise seems fine to me!

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]
Could we add the new `history/` directory to the linting check as well? Can be done by adding the following line to `.github/workflows/lint.yml`

```yaml
      - name: Lint with Flake8
        run: |
          pip install flake8
          flake8 --ignore=E501,E731 server
          flake8 --ignore=E501,E731 history
```

File: history/create_history_db.py (hunk 2 above)
Comment (idreyn) [reply history/create_history_db.py:16]: Yep, that's right.

File: history/util.py (hunk 1 above)
Comment (idreyn) [reply history/util.py:9]: Not at all. I'm a UUID fiend and I didn't check the existing schema, but I agree that `nextval` is simpler!

File: server/routes.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        return "Red-A" if "Ashmont" in route_pattern_name else "Red-B"
+    return default_route_id
+
+
+def derive_custom_direction_destinations(
+    route, normalized_route_name, custom_route_name
+):
+    if normalized_route_name == "Red":
+        if custom_route_name == "Red-A":
+            return ["Braintree", "Alewife"]
```
[the remark this answers is not in the document]
Comment (idreyn) [reply server/routes.py:22]: 😳

File: server/secrets.py (hunk 3 above)
Comment (idreyn) [reply server/secrets.py:3]: Oh, yeah, I thought I configured this to not commit my changes to secrets! (Luckily there are no actual secrets in here)

**Comment by idreyn** [comment]
> Could we add the new `history/` directory to the linting check as well?

Great catch — after thinking about it for a bit I think it's better if we just enforce that all Python code lives in `server` by convention, and I moved the history code back into that directory. I also added `flake8` to the Pipfile dev deps and I imagine I'll also eventually set up some pre-push hooks or something like that so I don't drive myself crazy pushing up unlinted code.

**Review by nathan-weinberg — APPROVED** [review]
LGTM!

**Pull request merged**